sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,39 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   */
+  object SubstituteHint extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case logical: LogicalPlan => logical transformDown {
+        // MAPJOIN
+        case h @ Hint(name, parameters, child) if name.equalsIgnoreCase("MAPJOIN") =>
+          var resolvedChild = child
+          for (table <- parameters) {
+            var stop = false
+            resolvedChild = child.transformDown {
+              case r @ BroadcastHint(UnresolvedRelation(_, _)) => r
+              case r @ UnresolvedRelation(t, _) if !stop && t.table == table =>
+                stop = true
+                BroadcastHint(r)
+            }
+          }
+          resolvedChild
+        // BROADCAST (LEFT|RIGHT)
+        case h @ Hint(name, parameters, j @ Join(left, _, _, _))
+            if name.equalsIgnoreCase("BROADCAST_LEFT") =>
+          j.copy(left = BroadcastHint(left))
+        case h @ Hint(name, parameters, j @ Join(_, right, _, _))
+          if name.equalsIgnoreCase("BROADCAST_RIGHT") =>
+          j.copy(right = BroadcastHint(right))
+
+        // Remove unrecognized hint
+        case Hint(_, _, child) => child
```

Comment:
I think we can add `logDebug` here.
But, usually, you know, DBMS ignores `Hint` by default.
Any malformed or unknown hints become **comment**.